`jsdoc/check-line-alignment` incorrectly triggered if `@returns` description starts with `-1`

It seems that the `jsdoc/check-line-alignment` rule is triggered if an `@returns` description begins with `-1` (number).

## Expected behavior
AFAIK, the dash is not recommended to delimit the @returns type from the description. So I would expect this to not trigger the error.

## Actual behavior

Error is output:

```
C:\dev\engineclean\src\scene\graph-node.js
  915:1  error  Expected JSDoc block lines to not be aligned  jsdoc/check-line-alignment

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

## ESLint Config

Include this:

```
"jsdoc/check-alignment": "error"
```

This appears in this full config:

https://github.com/playcanvas/playcanvas-eslint-config/blob/9dac541d64a73876ed27d5298eed7197981e6958/index.js#L400

## ESLint sample

This code triggers the error:

```js
    /**
     * Returns cached value of negative scale of the world transform.
     *
     * @returns {number} -1 if world transform has negative scale, 1 otherwise.
     * @ignore
     */
    get worldScaleSign() {

        if (this._worldScaleSign === 0) {
            this._worldScaleSign = this.getWorldTransform().scaleSign;
        }

        return this._worldScaleSign;
    }
```

Error output is:

![image](https://github.com/gajus/eslint-plugin-jsdoc/assets/697563/14bcfb43-0d8d-4fe8-80ed-bc8788275450)

Doing an autofix will update this:

```js
     * @returns {number} -1 if world transform has negative scale, 1 otherwise.
```

To this:

```js
     * @returns {number} - 1 if world transform has negative scale, 1 otherwise.
```

This completely changes the meaning of the docs, of course.

## Environment

- Node version: v18.13.0
- ESLint version: v8.41.0
- `eslint-plugin-jsdoc` version: v44.2.4

--- a/docs/rules/check-line-alignment.md
+++ b/docs/rules/check-line-alignment.md
@@ -991,5 +991,12 @@ function quux () {
  */
 function quux () {}
 // "jsdoc/check-line-alignment": ["error"|"warn", "never",{"wrapIndent":"   "}]
+
+/**
+ * Returns cached value of negative scale of the world transform.
+ *
+ * @returns {number} -1 if world transform has negative scale, 1 otherwise.
+ */
+// "jsdoc/check-line-alignment": ["error"|"warn", "never"]
 ````
 

--- a/docs/rules/require-param.md
+++ b/docs/rules/require-param.md
@@ -1812,5 +1812,9 @@ function foo(this: T, bar: number): number {
   console.log(this.name);
   return bar;
 }
+
+/** {@link someOtherval} */
+function a (b) {}
+// "jsdoc/require-param": ["error"|"warn", {"contexts":[{"comment":"*:not(JsdocBlock:has(JsdocInlineTag[tag=link]))","context":"FunctionDeclaration"}]}]
 ````
 

--- a/src/alignTransform.js
+++ b/src/alignTransform.js
@@ -306,7 +306,7 @@ const alignTransform = ({
     }
 
     const postHyphenSpacing = customSpacings?.postHyphen ?? 1;
-    const hyphenSpacing = /^\s*-\s*/u;
+    const hyphenSpacing = /^\s*-\s+/u;
     tokens.description = tokens.description.replace(
       hyphenSpacing, '-' + ''.padStart(postHyphenSpacing, ' '),
     );

--- a/src/rules/checkLineAlignment.js
+++ b/src/rules/checkLineAlignment.js
@@ -94,7 +94,7 @@ const checkNotAlignedPerTag = (utils, tag, customSpacings) => {
 
   const postHyphenSpacing = customSpacings?.postHyphen ?? 1;
   const exactHyphenSpacing = new RegExp(`^\\s*-\\s{${postHyphenSpacing},${postHyphenSpacing}}(?!\\s)`, 'u');
-  const hasNoHyphen = !(/^\s*-(?!$)/u).test(tokens.description);
+  const hasNoHyphen = !(/^\s*-(?!$)(?=\s)/u).test(tokens.description);
   const hasExactHyphenSpacing = exactHyphenSpacing.test(
     tokens.description,
   );
@@ -144,7 +144,7 @@ const checkNotAlignedPerTag = (utils, tag, customSpacings) => {
     }
 
     if (!hasExactHyphenSpacing) {
-      const hyphenSpacing = /^\s*-\s*/u;
+      const hyphenSpacing = /^\s*-\s+/u;
       tokens.description = tokens.description.replace(
         hyphenSpacing, '-' + ''.padStart(postHyphenSpacing, ' '),
       );

--- a/test/rules/assertions/checkLineAlignment.js
+++ b/test/rules/assertions/checkLineAlignment.js
@@ -2170,5 +2170,17 @@ export default {
         },
       ],
     },
+    {
+      code: `
+      /**
+       * Returns cached value of negative scale of the world transform.
+       *
+       * @returns {number} -1 if world transform has negative scale, 1 otherwise.
+       */
+      `,
+      options: [
+        'never',
+      ],
+    },
   ],
 };

--- a/test/rules/assertions/requireParam.js
+++ b/test/rules/assertions/requireParam.js
@@ -3525,5 +3525,21 @@ export default {
       `,
       parser: require.resolve('@typescript-eslint/parser'),
     },
+    {
+      code: `
+        /** {@link someOtherval} */
+        function a (b) {}
+      `,
+      options: [
+        {
+          contexts: [
+            {
+              comment: '*:not(JsdocBlock:has(JsdocInlineTag[tag=link]))',
+              context: 'FunctionDeclaration',
+            },
+          ],
+        },
+      ],
+    },
   ],
 };

